src/java/org/apache/cassandra/db/SSTableImporter.java

```diff
@@ -97,7 +99,19 @@ synchronized List<String> importNewSSTables(Options options)
                         try
                         {
                             abortIfDraining();
-                            verifySSTableForImport(descriptor, entry.getValue(), options.verifyTokens, options.verifySSTables, options.extendedVerify);
+
+                            if (options.failOnMissingIndex && cfs.indexManager.getIndexGroup(StorageAttachedIndexGroup.GROUP_KEY) != null)
```

Comment:
we also enter this `if` only if there are some SAI indexes for this table present ... obviously.